<launch>
    <!-- Global Front Facing Camera topic -->
    <arg name="camera_topic" default="/cam_pub/image_raw" />
    <!-- ......................................................................... -->


    <!-- rqt_reconfigure aka Dynamic Reconfigure GUI ............................. -->
    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_console" args="-t" />
    <!-- ......................................................................... -->


    <!-- DataSpeed DBW ........................................................... -->
    <include file="$(find dbw_polaris_can)/launch/dbw.launch">
        <arg name="laod_urdf" value="false" />
    </include>
    <!-- ......................................................................... -->


    <!-- USB Camera .............................................................. -->
    <node name="cam_pub" pkg="simple_camera_publisher" type="cam_pub" respawn="true" respawn_delay="10" output="screen">
        <!-- Use device path from /dev/v4l/by-id -->
        <param name="source" type="string" value="/dev/v4l/by-id/usb-S1F0009330LB620L4500072R_Integrated_Camera_SunplusIT_Inc-video-index0" />
        <param name="show_output" type="bool" value="true" />
    </node>
    <!-- ......................................................................... -->


    <!-- Piksi Multi GPS -->
    <!-- <include file="$(find piksi_multi_cpp)/launch/rover.launch">
        <arg name="ns" value="reference" />
        <arg name="device_ids" value="tcp://192.168.0.xxx:55555" />
    </include>

    <inlcude file="$(find piksi_multi_cpp)/launch/rover.launch">
        <arg name="ns" value="attitude" />
        <arg name="device_ids" value="tcp://192.168.0.xxx:55555" />
    </include> -->
    <!-- ......................................................................... -->


    <!-- DBScan Lane Following -->
    <node name="advanced_dbw_follow_lane" pkg="follow_line_pkg" type="advanced_dbw_follow_lane.py" required="true" output="screen">
        <param name="imgtopic_name" type="str" value="cam_pub/image_raw" />
    </node>
    <!-- ......................................................................... -->

    <!-- Vehicle Controller -->
    <node name="vc_node" pkg="vehicle_pkg" type="vc.py" required="true" output="screen">
        <param name="light_ns" type="string" value="north" />
    </node>
    <!-- ......................................................................... -->


</launch>